tests/requests/tests.py

```diff
@@ -830,3 +830,251 @@ def test_request_path_begins_with_two_slashes(self):
+            'Connection': 'keep-alive',
+            'Pragma': 'no-cache',
+            'Cache-Control': 'no-cache',
+            'Upgrade-Insecure-Requests': '1',
+            'Accept-Language': 'es-419,es;q=0.9,en;q=0.8,en-US;q=0.7',
+            'Cookie': '%7B%22hello%22%3A%22world%22%7D;another=value',
+            'X-Proto': 'https',
+            'X-Forwarded-Host': 'forward.com',
+            'X-Forwarded-Port': '80',
+            'X-Forwarded-Protocol': 'https',
+            'X-Custom-Header-1': 'custom_header_1',
+            'X-Custom-Header-2': 'custom_header_2'})
+
+    def test_wsgi_request_headers(self):
+        request = WSGIRequest(self.TESTING_ENVIRON)
+
+        self.assertEqual(dict(request.headers), {
+            'Content-Type': 'text/html',
+            'Content-Length': '100',
+            'Accept': '*',
+            'Host': 'example.com',
+            'User-Agent': 'python-requests/1.2.0',
+            'Referer': 'https://docs.djangoproject.com',
+            'If-Match': 'py7h0n',
+            'If-None-Match': 'dj4n60',
+            'If-Modified-Since': 'Sat, 12 Feb 2011 17:38:44 GMT',
+            'Accept-Encoding': 'gzip, deflate, br',
+            'Connection': 'keep-alive',
+            'Pragma': 'no-cache',
+            'Cache-Control': 'no-cache',
+            'Upgrade-Insecure-Requests': '1',
+            'Accept-Language': 'es-419,es;q=0.9,en;q=0.8,en-US;q=0.7',
+            'Cookie': '%7B%22hello%22%3A%22world%22%7D;another=value',
+            'X-Proto': 'https',
+            'X-Forwarded-Host': 'forward.com',
+            'X-Forwarded-Port': '80',
+            'X-Forwarded-Protocol': 'https',
+            'X-Custom-Header-1': 'custom_header_1',
+            'X-Custom-Header-2': 'custom_header_2'})
+
+    def test_wsgi_request_headers_getitem(self):
+        request = WSGIRequest(self.TESTING_ENVIRON)
+
+        self.assertEqual(request.headers['User-Agent'], 'python-requests/1.2.0')
+        self.assertEqual(
+            request.headers['X-Custom-Header-1'], 'custom_header_1')
+        self.assertEqual(request.headers['Pragma'], 'no-cache')
+        self.assertEqual(request.headers['Content-Type'], 'text/html')
+        self.assertEqual(request.headers['Content-Length'], '100')
+
+    def test_wsgi_request_headers_get(self):
+        request = WSGIRequest(self.TESTING_ENVIRON)
+
+        self.assertEqual(
+            request.headers.get('User-Agent'), 'python-requests/1.2.0')
+        self.assertEqual(
+            request.headers.get('X-Custom-Header-1'), 'custom_header_1')
+        self.assertEqual(request.headers.get('Pragma'), 'no-cache')
+        self.assertEqual(request.headers.get('Content-Type'), 'text/html')
+        self.assertEqual(request.headers.get('Content-Length'), '100')
+
+
+class EnvironHeadersTests(SimpleTestCase):
+    def test_parse_cgi_headers_basics(self):
+        parser = EnvironHeaders.parse_cgi_header_name
+
+        self.assertEqual(parser('HTTP_ACCEPT'), 'Accept')
```

Comment:
Similar to my "use a simpler dictionary" comment above, this fells like overkill. The logic is fairly simple. What does "'HTTP_X_CUSTOM_HEADER_1" test that "HTTP_X_CUSTOM_HEADER_2" doesn't? Also you can use a loop with `subTest()` to simplify rather than have separate methods. For example:
https://github.com/django/django/blob/f021c110d02fd7ca32ae56f511b46e5d138b6c73/tests/utils_tests/test_http.py#L156-L158